### DISCUSSION TITLE
help finding reset button connections

### DISCUSSION THREAD

**Discussion by jimmypikachu** [arendst/Tasmota#18300] [Q&A]
Hi, I had successfully flashed many sonoffs to tasmota back in 2019 however things might have changed a bit now.

I bought  a 1 gang switch, on the box it is labelled TX T2UK1C, and on the board it read "T1UK TOUCH V1.0 2018.10.17"

If I simply poweron by connecting the VCC TX RX GND, the blue wifi LED starts to blink
If I pull down R19 while power on (I seem to remember this is the procedure to flash it), the blue LED does not blink, however when I try flashing it with tasmota.bin, it just do below

        esptool v0.4.12 - (c) 2014 Ch. Klippel <ck@atelier-klippel.de>
	setting board to nodemcu
	setting baudrate from 115200 to 115200
	setting port from  to COM5
	setting address from 0x00000000 to 0x00000000
	espcomm_upload_file
	espcomm_upload_mem
	setting serial port timeouts to 1000 ms
        opening bootloader
        resetting board
        trying to connect
	flush start
	setting serial port timeouts to 1 ms
	setting serial port timeouts to 1000 ms
	flush complete
	espcomm_send_command: sending command header
	espcomm_send_command: sending command payload
	read 0, requested 1
        trying to connect
	flush start
	setting serial port timeouts to 1 ms
	setting serial port timeouts to 1000 ms
	flush complete
	espcomm_send_command: sending command header
	espcomm_send_command: sending command payload
	read 0, requested 1
        warning: espcomm_sync failed
        error: espcomm_open failed
        error: espcomm_upload_mem failed
        STOPPED due to 2 errors! (try reset on the unit, then start a new flash attempt)

I believe it is not yet in flash mode, I read from https://tasmota.github.io/docs/devices/Sonoff-T1/#touch-board-ver1 seems the reset button should also be involved, I definitely remember my old sonoff tasmota T1 had that reset button, but in this TX version it is not present. Instead in the original location, there is a R23 resistor and R29 with 2 pads of solder without any resistor.

May I get help for someone who owns a T1 to find out which connections need to be shorted to simulate the reset button press?

**Comment by sfromis** [comment]
For those devices, you need no "reset button", but instead grounding gpio 0 while powering up for flashing. Instead of trying to find a "good resistor" to connect to, I'd recommend grounding gpio 0 directly on the ESP chip, bottom 2nd from the right on the pinout:
https://tasmota.github.io/docs/Pinouts/#esp8266-and-esp8285

Don't worry about the pins being tiny, you can safely also ground gpio 4 (to the right of gpio 0) at the same time, meaning that you can rest a pin between those two while applying power, and then proceed to flash.

**Reply by jimmypikachu** [reply]
thanks so much, I will try that
